host/src/connection_manager.rs

```diff
@@ -387,7 +411,34 @@ impl<'d> ConnectionManager<'d> {
         Poll::Ready(Err(Error::NotFound))
     }
 
-    pub(crate) fn get_att_mtu(&self, conn: ConnHandle) -> u16 {
+    pub(crate) fn get_att_mtu(&self, index: u8) -> u16 {
+        self.with_mut(|state| state.connections[index as usize].att_mtu)
+    }
+
+    pub(crate) async fn send(&self, index: u8, pdu: Pdu<'d>) {
+        let handle = self.with_mut(|state| state.connections[index as usize].handle.unwrap());
```

Comment:
It has some implications on stack/RAM usage because now you'd need to create a "big" ConnectionStorage when allocing, rather than just modifying one field. It's a while ago, so I don't have the numbers. 

But it could be worth trying again, and measuring size change.

Reply:
Interesting, I'd have expected the compiler to optimise to modify-in-place if the array already exists, but maybe I'm not fully grasping the way it's used.